main/src/ca/uwaterloo/flix/language/phase/optimizer/Inliner.scala

```diff
@@ -181,18 +181,17 @@ object Inliner {
           Expr.ApplyClo(e1, e2, tpe, eff, loc)
       }
 
-    case Expr.ApplyDef(sym, exps, itpe, tpe, eff, loc) =>
-      val es = exps.map(visitExp(_, ctx0))
-      Expr.ApplyDef(sym, es, itpe, tpe, eff, loc)
+    case exp@Expr.ApplyDef(_, _, _, _, _, _) =>
+      callSiteInlineDef(exp, ctx0)
```

Comment:
Can you move the code here?

Currently call this helper function just to pattern match again. We never use that pattern, instead we aim for structurally recursive functions.